src/azure-cli/azure/cli/command_modules/appconfig/_featuremodels.py

```diff
@@ -50,36 +50,71 @@ class FeatureFlagValue:
         Represents if the Feature flag is On/Off/Conditionally On
     :ivar dict {string, FeatureFilter[]} conditions:
         Dictionary that contains client_filters List (and server_filters List in future)
-    '''
-
-    def __init__(self,
-                 id_,
-                 description=None,
-                 enabled=None,
-                 conditions=None,
-                 display_name=None):
+    :ivar FeatureAllocation allocation:
+        Determines how variants should be allocated for the feature to various users.
+    :ivar list FeatureVariant[] variants:
+        The list of variants defined for this feature.
+    :ivar dict telemetry:
+        The declaration of options used to configure telemetry for this feature.
+    """
+
+    def __init__(
+        self,
+        id_,
+        description=None,
+        enabled=None,
+        conditions=None,
+        display_name=None,
+        variants=None,
+        allocation=None,
+        telemetry=None,
+    ):
         default_conditions = {FeatureFlagConstants.CLIENT_FILTERS: []}
 
         self.id = id_
         self.description = "" if description is None else description
         self.enabled = enabled if enabled else False
         self.conditions = conditions if conditions else default_conditions
-        self.display_name = display_name
+        if allocation is not None:
+            self.allocation = allocation
+        if variants is not None:
+            self.variants = variants
+        if telemetry is not None:
+            self.telemetry = telemetry
+        if display_name is not None:
+            self.display_name = display_name
 
     def __repr__(self):
         featureflagvalue = {
             FeatureFlagConstants.ID: self.id,
             FeatureFlagConstants.DESCRIPTION: self.description,
             FeatureFlagConstants.ENABLED: self.enabled,
-            FeatureFlagConstants.CONDITIONS: custom_serialize_conditions(self.conditions),
-            FeatureFlagConstants.DISPLAY_NAME: self.display_name
+            FeatureFlagConstants.CONDITIONS: custom_serialize_conditions(
+                self.conditions
+            ),
         }
 
+        if hasattr(self, FeatureFlagConstants.DISPLAY_NAME):
+            featureflagvalue[FeatureFlagConstants.DISPLAY_NAME] = self.display_name
+
+        if hasattr(self, FeatureFlagConstants.ALLOCATION):
+            featureflagvalue[FeatureFlagConstants.ALLOCATION] = (
+                custom_serialize_allocation(self.allocation)
+            )
```

Comment:
Is there a reason for the extra brackets?

Reply:
No need for this. Removed